plutus-core/plutus-core/test/Evaluation/Spec.hs

```diff
@@ -1,11 +1,117 @@
-module Evaluation.Spec where
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE LambdaCase            #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PolyKinds             #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE TypeApplications      #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# LANGUAGE TypeOperators         #-}
 
-import Evaluation.Machines (test_machines)
+module Evaluation.Spec (test_evaluation) where
+
+import PlutusCore hiding (Term)
+import PlutusCore qualified as PLC
+import PlutusCore.Builtin
+import PlutusCore.Generators (forAllNoShow, genValArg)
+import PlutusCore.Generators.AST
+import PlutusCore.Pretty
 
+import Control.Exception
+import Control.Monad.Except
+import Control.Monad.Extra
+import Data.Functor (($>))
+import Data.List.Extra qualified as List
+import Evaluation.Machines (test_machines)
+import Hedgehog hiding (Opaque, Var, eval)
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Internal.Property (failWith)
 import Test.Tasty
+import Test.Tasty.Hedgehog
+import Type.Reflection
+
+type Term = PLC.Term TyName Name DefaultUni DefaultFun ()
+
+test_builtinsDon'tThrow :: TestTree
+test_builtinsDon'tThrow =
+    testGroup
+        "Builtins don't throw"
+        $ fmap (\fun -> testProperty (display fun) $ prop_builtinsDon'tThrow fun) List.enumerate
+
+prop_builtinsDon'tThrow :: DefaultFun -> Property
+prop_builtinsDon'tThrow bn = property $ do
+    args <- forAllNoShow . Gen.choice $ [genArgsWellTyped bn, genArgsArbitrary bn]
+    mbErr <-
+        liftIO $
+            catch
+                (($> Nothing) . evaluate . runEmitter . runExceptT $ eval args)
+                (pure . pure)
+    whenJust mbErr $ \(e :: SomeException) -> do
+        let msg =
+                "Builtin function evaluation failed"
```

Comment:
I think you can do this more simply with annotate: https://hackage.haskell.org/package/hedgehog-0.6.1/docs/Hedgehog.html#v:annotate

i.e. do something like:
```
annotate $ "Function: " <> display bn
annotate $ "Arguments: " <> ...
...
```
and then ditch the exception handler. I think what will happen then is that the exception will fail the test and Hedgehog will show you the annotated values in the output.

Reply:
I changed to using `annotate`, but didn't remove the exception handler because the annotations don't show up when the test is failed by an exception.